[JIT] adding missing test cases for test_isinstance.py

lobstergrindset opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47396 [JIT] adding missing test cases for test_isinstance.py**

Adds tests for missing functionality exposed in issue #37516 and resolved in PR #46130

Will not pass tests until #46130 is merged 

Differential Revision: [D24739765](https://our.internmc.facebook.com/intern/diff/D24739765)

Commit:
[JIT] adding missing test cases for test_isinstance.py

[ghstack-poisoned]

SplitInfinity reviewed (approved):
Cool, make sure to wait for that other PR to land first (are there plans to land it soon?)

Lilyjjo commented:
will do, and it looks like Elias has been trying to land it based off his activity with it on Phabriactor

Commit:
Update on "[JIT] adding missing test cases for test_isinstance.py"


Adds tests for missing functionality exposed in issue #37516 and resolved in PR #46130

Will not pass tests until #46130 is merged 

Differential Revision: [D24739765](https://our.internmc.facebook.com/intern/diff/D24739765)

[ghstack-poisoned]
